lib/services/workflow-api-service.js

```diff
@@ -128,7 +128,56 @@ function workflowApiServiceFactory(
     };
 
     WorkflowApiService.prototype.cancelTaskGraph = function(graphId) {
-        return taskGraphProtocol.cancelTaskGraph(graphId);
+        var instanceId;
+
+        return waterline.graphobjects.needByIdentifier(graphId)
+        .then(function(workflow) {
+            instanceId = workflow.instanceId;
+
+            if (workflow.active()) {
+                return taskGraphProtocol.cancelTaskGraph(instanceId);
+            }
+            throw new Errors.TaskCancellationError(
+                graphId + ' is not an active workflow'
+            );
```

Comment:
This is another case for my above concern about putting error handling into branch. But this case is a little nicer than above because it is not in the end of the function.